CrossRef-REST.js

```diff
@@ -0,0 +1,245 @@
+				.includes(result.type)) {
+			item = new Zotero.Item("bookSection");
+			item.publisher = result.publisher;
+			item.place = result['publisher-location'];
+			if (result.ISBN) item.ISBN = result.ISBN[0];
+		}
+		// Report
+		else if (['dataset', 'posted-content', 'report', 'report-series', 'standard']
+				.includes(result.type)) {
+			item = new Zotero.Item("report");
+			item.institution = result.publisher;
+			item.place = result['publisher-location'];
+			item.seriesTitle = result['container-title'];
+		}
+		// Conference paper
+		else if (['proceedings-article'].includes(result.type)) {
+			item = new Zotero.Item("conferencePaper");
+			item.proceedingsTitle = result['container-title'];
+			item.publisher = result.publisher;
+			if (result.event) {
+				item.conferenceName = result.event.name;
+				item.place = result.event.location;
+			}
+			if (result.ISBN) item.ISBN = result.ISBN[0];
+		}
+		// Thesis
+		else if (['dissertation'].includes(result.type)) {
+			item = new Zotero.Item("thesis");
+			item.university = result.publisher;
+			item.place = result['publisher-location'];
+		}
+		else {
+			return;
+		}
+		
+		// edited-book, standard-series - ignore, because CrossRef has zero results for this type
+		// component, journal, journal-issue, journal-volume, other, proceedings - ignore,
+		// because Zotero doesn't have equivalent item types.
+		
+		item.abstractNote = result.abstract;
+		
+		parseCreators(result, item);
+		
+		
+		// Contains the earliest of: published-online, published-print, content-created
+		let pubDate = result['issued'];
+		
+		if (pubDate && pubDate['date-parts'][0]) {
+			let year = pubDate['date-parts'][0][0];
+			let month = pubDate['date-parts'][0][1];
+			let day = pubDate['date-parts'][0][2];
+			
+			if (year) {
+				if (month) {
+					if (day) {
+						item.date = year + "-" + month + "-" + day;
+					}
+					else {
+						item.date = month + "/" + year;
+					}
+				}
+				else {
+					item.date = year;
+				}
+			}
+		}
+		
+		item.pages = result.page;
+		
+		if (result.DOI) {
+			if (ZU.fieldIsValidForType('DOI', item.itemType)) {
+				item.DOI = result.DOI;
+			}
+			// add DOI to extra for unsupprted items
+			else {
+				if (item.extra) {
+					item.extra += '\nDOI: ' + result.DOI;
+				}
+				else {
+					item.extra = 'DOI: ' + result.DOI;
+				}
+			}
+		}
+		
+		// result.URL is always http://dx.doi.org/..
+		
+		if (result.link && result.link.URL) item.url = result.link.URL;
+		
+		if (result.title && result.title[0]) {
+			item.title = result.title[0];
+			if (result.subtitle && result.subtitle[0]) {
+				// Don't duplicate subtitle if it already exists in title
+				if (item.title.toLowerCase().indexOf(result.subtitle[0].toLowerCase()) < 0) {
+					item.title += ': ' + result.subtitle[0];
+				}
+			}
+			item.title = item.title.replace(/<!\[CDATA\[/g, '');
+			item.title = item.title.replace(/\]\]>/g, '');
+			item.title = item.title.replace(/<[a-zA-Z\/].*?>/g, '');
```

Comment:
What about supported/transformed markup, [as in the old CrossRef translator](https://github.com/zotero/translators/blob/c51821e26a97a1371730efdec17c80d83b0cef02/CrossRef.js#L40)? While currently we don't officially support markup in titles, citeproc-js does render it, so we don't want to throw away fields that citeproc-js supports.